docs/ShowInDialogButton.md

```diff
@@ -5,129 +5,301 @@ title: "ShowInDialogButton"
-      <CustomerLayout />
-    </ShowInDialogButton>
-  );
-
-  return (
-    <ReferenceManyField
-      label="Customers"
-      reference="customers"
-      target="employer_id"
-    >
-      <Datagrid>
-        <TextField source="id" />
-        <TextField source="first_name" />
-        <TextField source="last_name" />
-        <DateField source="dob" label="born" />
-        <SelectField source="sex" choices={sexChoices} />
-        {/* Using a component as label is a trick to render it in the Datagrid header */}
-        <DatagridActionsColumn label={createButton}>
-          {editButton}
-          {showButton}
-        </DatagridActionsColumn>
-      </Datagrid>
-    </ReferenceManyField>
-  );
-};
-
 const EmployerEdit = () => (
   <Edit>
     <SimpleForm>
       <TextInput source="name" validate={required()} />
       <TextInput source="address" validate={required()} />
       <TextInput source="city" validate={required()} />
-      <NestedCustomersDatagrid />
+      <ReferenceManyField label="Customers" reference="customers" target="employer_id">
+        <CustomersDatagrid />
+      </ReferenceManyField>
     </SimpleForm>
   </Edit>
 );
 ```
 {% endraw %}
 
-Check out [the `ra-form-layout` documentation](https://marmelab.com/ra-enterprise/modules/ra-form-layout#createindialogbutton-editindialogbutton-and-showindialogbutton) for more details.
+## Props
+
+This component accepts the following props:
+
+| Prop           | Required | Type           | Default  | Description               |
+|----------------|----------|----------------|----------|---------------------------|
+| `children`     | Required | `ReactNode`    |          | The content of the dialog |
+| `ButtonProps`  | Optional | `object`       |          | Props to pass to the MUI `<Button>` component |
+| `empty WhileLoading` | Optional | `boolean` |         | Set to `true` to return `null` while the list is loading |
+| `fullWidth`    | Optional | `boolean`      | `false`  | Set to `true` to make the dialog full width |
+| `icon`         | Optional | `ReactElement` |          | The icon of the button |
+| `id`           | Optional | `string | number` |       | The record id. If not provided, it will be deduced from the record context |
+| `inline`       | Optional | `boolean`      | `false`  | Set to `true` to display only an MUI `<IconButton>` instead of the full `<Button>`. The label will still be available as a `<Tooltip>` though. |
+| `label`        | Optional | `string`       |          | The label of the button. I18N is supported. |
+| `maxWidth`     | Optional | `string | boolean` | `sm` | The max width of the dialog. |
+| `queryOptions` | Optional | `object`       |          | The options to pass to the `useQuery` hook |
+| `resource`     | Optional | `string`       |          | The resource name, e.g. `posts` |
+| `sx`           | Optional | `object`       |          | Override the styles applied to the dialog component |
```

Comment:
Should we report all these improvements to the doc in the EE doc as well? Currently, `<CreateInDialogButton>`, `<EditInDialogButton>` and `<ShowInDialogButton>` are documented in a single section. Should we split?

Reply:
nice to have. We don't reference the EE doc from this page